src/Microsoft.DotNet.Tools.Compiler.Native/IntermediateCompilation/Linux/LinuxCppCompileStep.cs

```diff
@@ -70,7 +70,8 @@ private void InitializeArgs(NativeCompileSettings config)
             //
             // Get the directory name to ensure there are no trailing slashes as they may conflict
             // with the terminating "  we suffix to account for paths with spaces in them.
-            var ilcSdkIncPath = Path.GetDirectoryName(config.IlcSdkPath);
+            char[] charsToTrim = {'\\', '/'};
+            var ilcSdkIncPath = config.IlcSdkPath.TrimEnd(charsToTrim);
```

Comment:
Single quote doesn't treat \\ as escape char on Unix. Isn't that the problem? Didn't understand why having trailing / has an issue?